File: bubble/bubble.py
#!/usr/bin/python
# Using python 3.9

import random

# Generate random numbers to sort
def genNumbers(size):
	numbers = [random.randint(0,9) for x in range(size)]
	return numbers

def bubbleSort(numbers):
	n = len(numbers)	
	count = 0
	for i in range(n):
		for j in range(n-i-1):
			#print(numbers)
			if numbers[j] > numbers[j+1]:
				count = count + 1
				numbers[j], numbers[j+1] = numbers[j+1], numbers[j]
	print("sort took " + str(count) + " operations to complete")
	return numbers
unsortedNumbers = genNumbers(20)
print("Unsorted: " + str(unsortedNumbers))
sortedNumbers = bubbleSort(unsortedNumbers)
print("Sorted: " + str(sortedNumbers))

File: bubble/bubble.cpp
#include <iostream>
using namespace std;

void genNumbers(int array[], int size){
	for (int i=0; i<size; i++){
		array[i] = rand() % size;
	}
}

void bubbleSort(int numbers[], int size) {
	for (int i=0; i < size; i++){
		for(int j=0; j < size; j++){
			if (numbers[i] > numbers[j]){
				int tmp = numbers[i];
				numbers[i] = numbers[j];
				numbers[j] = tmp;
			}
		}
	}
}

void printNumbers(int numbers[], int size){
	for(int i=0; i < size; i++){
		cout << numbers[i] << " ";
	}
	cout << endl;
}

int main(){
	srand((int)time(0));

	int size = 10;
	int numbers[size];		
	genNumbers(numbers, size);
	
	cout << "unsorted: ";	
  printNumbers(numbers, size);

	bubbleSort(numbers, size);
	
	cout << "sorted: ";
	printNumbers(numbers, size);
}
